File: src/model/lecture.model.js
import mongoose, { Schema } from "mongoose";

const lectureSchema = new mongoose.Schema(
  {
    byTeacher: {
      type: Schema.Types.ObjectId,
      ref: "teachers",
      required: true,
    },
    adminEmail:{
      type:String,
      trim:true,
      required:true
    },
    lectureName: {
      type: String,
      required: true,
    },
    standard: {
      type: String,
      required: true,
    },
    date: {
      type: String,
      required: true,
      default:"Everyday"
    },
    startTime: {
      type: String,
      required: true,
    },
    endTime: {
      type: String,
      required: true,
    },
    description: {
      type: String,
    },
    board: {
      type: String,
      required: true,
      uppercase:true
    },
  },
  { timestamps: true }
);

export const lectures = new mongoose.model(
  "lectures",
  lectureSchema
);

File: class_frontend/lib/services/models/lecture.model.dart
class Lecture {
  final String id;
  final String standard;
  final String lectureName;
  final String teacherName;
  final String time;
  final String description;
  final String board;
  final String adminName;
  final String date;

  Lecture({
    required this.id,
    required this.standard,
    required this.lectureName,
    required this.teacherName,
    required this.time,
    required this.description,
    required this.board,
    required this.adminName,
     required this.date,
  });

  factory Lecture.fromJson(Map<String, dynamic> json) {
    return Lecture(
      id: json['_id'],
      standard: json['standard'],
      lectureName: json['lectureName'],
      teacherName: json['byTeacher']['name'],
      time: json['time'],
      description: json['description'],
      board: json['board'],
      adminName: json['adminName'],
      date: json['date'],
    );
  }
}
